learning_resources/migrations/0070_learningresource_location.py

```diff
@@ -0,0 +1,25 @@
+# Generated by Django 4.2.16 on 2024-09-24 15:28
+
+import django.contrib.postgres.fields
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("learning_resources", "0069_learningresource_ocw_topics"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="learningresource",
+            name="location",
+            field=django.contrib.postgres.fields.ArrayField(
+                base_field=models.CharField(max_length=256), null=True, size=None
```

Comment:
It will also make it heard to catch if some locations are duplicates like "Cambridge, MA" vs "Cambridge MA" with no comma - although we don't have a good way of not creating duplicates anyway if Sloan lists the same locations as different strings

Reply:
A learning resource run will only have 1 location.  But since different runs may have different locations, I decided to make the learning resource location field an array of all the run values.  But I can change it so that it is also a string field equal to the location of the next/best run, similar to availability and next_start_date, and calculated every night.  I'm not too keen on a many-to-many relation between courses/runs and a new location model, since location is just a string.

Reply:
Can this just be a string array field on learning resource then? I think my objection is having ['Cambridge, MA', 'Lausanne, Switzerland'] and just ['Cambridge, MA'] as seperate location records that multiple courses join to is bad

Reply:
Also could ['Lausanne, Switzerland', 'Cambridge, MA'] potentially be created as a separate location record from ['Cambridge, MA', 'Lausanne, Switzerland']

Reply:
> Can this just be a string array field on learning resource then

Do you mean `"Cambridge, MA; Lausanne, Switzerland"` instead of `['Cambridge, MA', 'Lausanne, Switzerland']` for `LearningResource.location`?  I could do that, or just set it to the one location of the next/best run.  The location array is sorted in the ETL pipeline so there shouldn't be a `['Lausanne, Switzerland', 'Cambridge, MA']`, it should always be `['Cambridge, MA', 'Lausanne, Switzerland']` or `'Cambridge, MA; Lausanne, Switzerland'`

I can go ahead and make location a separate model with just a location charfield, with `LearningResourceRun` having a relation to just one `Location` object and `LearningResource` having either the same (identical to that of the next/best run) or a many-to-many relation (all run locations).  But it's more joins on queries.

Reply:
Refactored so that location is a `CharField` for both, and its value for `LearningResource` is set to the same as the next/best `LearningResourceRun` in `learning_resources.etl.loaders.load_run_dependent_values`